src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -54,63 +48,54 @@ public final void start(Map<String, String> config) {
         pollDelayMs = (Long) parsedConfig.get(MarkLogicSourceConfig.WAIT_TIME);
-        this.constraintColumn = (String) parsedConfig.get(MarkLogicSourceConfig.CONSTRAINT_COLUMN_NAME);
-        if (!StringUtils.hasText(constraintColumn)) {
-            constraintColumn = null;
-        }
 
-        queryContextBuilder = AbstractRowBatchBuilder.newQueryContextBuilder(dataMovementManager, parsedConfig);
+        queryHandler = QueryHandler.newQueryHandler(databaseClient, parsedConfig);
+        rowBatcherBuilder = AbstractRowBatchBuilder.newRowBatcherBuilder(dataMovementManager, parsedConfig);
         logger.info("Started RowBatcherSourceTask");
     }
 
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
         // Temporary logging while testing CP to ensure the correct output format is being used
-        logger.info("Polling; QueryContextBuilder: " + queryContextBuilder.getClass().getName());
+        logger.info("Polling ; RowBatcherBuilder: " + rowBatcherBuilder.getClass().getName());
+        logger.info("Polling; QueryHandler: " + queryHandler.getClass().getName());
 
         List<SourceRecord> newSourceRecords = new Vector<>();
         logger.info("Temporary log statement for testing; sleeping for " + pollDelayMs + "ms");
         Thread.sleep(pollDelayMs);
 
         long start = System.currentTimeMillis();
         try {
-            loadNewQueryContext(newSourceRecords);
-       } catch (Exception ex) {
+            getNewRowBatcher(newSourceRecords);
+            performPoll();
+        } catch (Exception ex) {
             if (!rowBatcherErrorIsKnownServerBug(ex)) {
                 logger.error("Unable to poll for source records. Unable to initialize row batcher; cause: " + ex.getMessage());
             }
             return null;
         }
 
-        performPoll();
         logger.info("Source record count: " + newSourceRecords.size() + "; duration: " + (System.currentTimeMillis() - start));
         return newSourceRecords.isEmpty() ? null : newSourceRecords;
     }
 
-    protected void loadNewQueryContext(List<SourceRecord> newSourceRecords) {
-        queryContext = queryContextBuilder.newQueryContext(newSourceRecords, previousMaxConstraintColumnValue);
-    }
-
     protected void performPoll() {
         try {
             logger.info("Starting job");
-            RowBatcher<?> rowBatcher = queryContext.getRowBatcher();
             dataMovementManager.startJob(rowBatcher);
             logger.info("Awaiting completion");
             rowBatcher.awaitCompletion();
             dataMovementManager.stopJob(rowBatcher);
 
-            if (constraintColumn != null) {
-                long queryStartTimeInMillis = rowBatcher.getServerTimestamp();
-                previousMaxConstraintColumnValue = getBatchMaxValue(queryContext.getCurrentQuery(), queryStartTimeInMillis, constraintColumn);
-            }
+            long queryStartTimeInMillis = rowBatcher.getServerTimestamp();
```

Comment:
The Task knows it only needs to do this if a constraint column option has been set, so I think it should do an `if` statement here to only call this stuff if that option has been set.